app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsRepository.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.components.notifications
+
+import android.util.Log
+import androidx.paging.InvalidatingPagingSourceFactory
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.PagingSource
+import com.keylesspalace.tusky.entity.Notification
+import com.keylesspalace.tusky.network.MastodonApi
+import kotlinx.coroutines.flow.Flow
+import okhttp3.ResponseBody
+import retrofit2.Response
+import javax.inject.Inject
+
+class NotificationsRepository @Inject constructor(
```

Comment:
In what way does this store things ("repository")?

Is there somewhere some "caching" between app run times involved? (I don't see it here.)

Reply:
The `...Repository` classes do not store things. They are the mediation layer between the view models above them and the paging sources / remote mediators below them.

This is introduced in https://developer.android.com/topic/architecture#data-layer and described in more detail in https://developer.android.com/topic/architecture/data-layer